k9mail-library/src/main/java/com/fsck/k9/mail/store/imap/ImapFolderPusher.java

```diff
@@ -504,6 +503,12 @@ private int processUntaggedResponse(long oldMessageCount, ImapResponse response,
                             Timber.d("Got untagged EXPUNGE for msgseq %d for %s", msgSeq, getLogId());
                         }
 
+                        if (K9MailLib.isDebug()) {
+                            Timber.d("Scheduling removal of message with seq %d due to expunge", msgSeq);
+                        }
+
+                        removeMsgUids.add("thisisahack");
```

Comment:
I've removed the code changes in `ImapFolderPusher` and will add that in a later commit.